### PR TITLE
Saves ENR to disk on every address update

### PR DESCRIPTION
**Pull request by b-m-f**
## Issue Addressed

https://github.com/sigp/lighthouse/issues/434

## Proposed Changes

- Extract saving ENR to disk into its own function
- Save ENR to disk on every adress update

## Additional Info

- The dir to save the ENR to is taken from the NetworkConfig, and saved into the Discovery object on instantiation

### CODE REVIEW

**Comment by AgeManning** [comment]
Hey! This looks good. I'm away at the moment and should have more time on monday. I'll do a full review then :)

File: beacon_node/eth2-libp2p/src/discovery.rs [hunk 1]
```diff
@@ -53,6 +54,9 @@ pub struct Discovery<TSubstream> {
 
     /// Logger for the discovery behaviour.
     log: slog::Logger,
+
+    /// directory to save ENR to
+    enr_dir: String,
```
Comment (AgeManning) [inline beacon_node/eth2-libp2p/src/discovery.rs]:
```suggestion
    enr_dir: PathBuf,
```

File: beacon_node/eth2-libp2p/src/discovery.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ use slog::{debug, info, o, warn};
 use std::collections::HashSet;
 use std::fs::File;
 use std::io::prelude::*;
+use std::path::Path;
```
Comment (AgeManning) [inline beacon_node/eth2-libp2p/src/discovery.rs]:
```suggestion
use std::path::{Path, PathBuf};
```

File: beacon_node/eth2-libp2p/src/discovery.rs [hunk 3]
```diff
@@ -66,6 +70,11 @@ impl<TSubstream> Discovery<TSubstream> {
         // checks if current ENR matches that found on disk
         let local_enr = load_enr(local_key, config, &log)?;
 
+        let enr_dir = match config.network_dir.to_str() {
```
Comment (AgeManning) [inline beacon_node/eth2-libp2p/src/discovery.rs]: Should be able to remove this

File: beacon_node/eth2-libp2p/src/discovery.rs [hunk 4]
```diff
@@ -90,6 +99,7 @@ impl<TSubstream> Discovery<TSubstream> {
             tcp_port: config.libp2p_port,
             discovery,
             log,
+            enr_dir,
```
Comment (AgeManning) [inline beacon_node/eth2-libp2p/src/discovery.rs]:
```suggestion
            enr_dir: PathBuf::from(config.network_dir.clone()),,
```

File: beacon_node/eth2-libp2p/src/discovery.rs [hunk 5]
```diff
@@ -205,6 +215,9 @@ where
                             info!(self.log, "Address updated"; "IP" => format!("{}",socket.ip()));
                             let mut address = Multiaddr::from(socket.ip());
                             address.push(Protocol::Tcp(self.tcp_port));
+                            let enr = self.discovery.local_enr();
+                            save_enr_to_disc(Path::new(&self.enr_dir), enr, &self.log);
```
Comment (AgeManning) [inline beacon_node/eth2-libp2p/src/discovery.rs]:
```suggestion
                            save_enr_to_disc(&self.enr_dir, enr, &self.log);
```

**Comment by b-m-f** [comment]
Seems much neater, cheers!

Path and PathBuf really work nice together with the dereferencing.

**Comment by AgeManning** [comment]
Merged to master. Thanks :)

**Pull request closed without merging**